src/OmniSharp.Roslyn.VisualBasic/Services/Structure/BlockStructureService.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Composition;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using OmniSharp.Extensions;
+using OmniSharp.Mef;
+using OmniSharp.Models.V2;
+using OmniSharp.Services;
+using OmniSharp.Utilities;
+
+namespace OmniSharp.Roslyn.VisualBasic.Services.Structure
+{
+    [OmniSharpHandler(OmniSharpEndpoints.V2.BlockStructure, LanguageNames.VisualBasic)]
+    public class BlockStructureService : IRequestHandler<BlockStructureRequest, BlockStructureResponse>
```

Comment:
The service code is an exact copy of the C# version for now. I may extract a base abstract to share common code later.

Reply:
This is great!

The issue of how to structure the two languages together already comes up here, as you point out.

As a principle, I would suggest that trying to minimize changes to the existing OmniSharp code files is better than reducing duplicate code, when there is a choice. The reason for this is that I think we should try to stay in sync with OmniSharp as it changes, and merging commits from there will be much easier if we are not interfering with the existing files too much. So, I would say you should rather keep the kind of duplication you have here, and not try to extract common base classes, even when is means there is more duplication.

I guess the right way to do it would be to rather submit some abstractions into common code back to OmniSharp as a PR, and then pull it in from that way to refactor. But I imagine they would not want to bother with that now - not sure. Hopefully later, and then one can clean up and refactor better.

The approach you're taking in some places for the tests, where you add an optional language parameter, is fine too, because the change is at least small. But I would suggest you keep thinking about how the changes you make will interact with potential changes made on the OmniSharp side, which we want to merge in.

There's no easy rule, and these are just my thoughts, but in this case the 'best' code structure for the combined codebase might not be the 'best' code structure for a parallel fork that we want to maintain in sync, perhaps to combine and refactor later.

Reply:
@govert If the original OmniSharp maintainers accepts to adding VB support by community PRs. The overall process will be much easier. We should try to get in touch with them for doing so.

If they refused adding VB support currently, then we'll have to continue on this fork and have the cost of maintaining the fork in sync with the original repo 😞 

I think I'll open this PR against the original repo and see what they will say.

Reply:
I agree that it would be much better to have it all in OmniSharp, and is certainly how I would prefer it.

But I think the topic is VB is sensitive inside the dotnet , Roslyn and VS teams.
The fact that there is no support for VB in VS Code is not an accident (there is COBOL support).

So I think you should be cautious in how you approach it, and make sure that a PR will be welcome before you open it.
I fear if you just you open a PR with VB in the title, it might get the wrong attention.
Maybe a new discussion, or even better direct communication with someone working on it or on the slack or gitter channel would be better.

You might also ask them whether PRs that refactor the OmniSharp code (extract some base classes that can be shared) will be considered, even if they will not involve VB explicitly.

Overall I'm saying it would be a great outcome to just have the OmniSharp project support both languages, but is more likely to be successful if you are a bit cautious.

Reply:
And I should say that, judging by the response to my attempts to raise the topic of VB here
https://github.com/OmniSharp/omnisharp-vscode/issues/25#issuecomment-665672039
and the 'scripting' topic here
https://github.com/dotnet/roslyn/discussions/50584
and here
https://github.com/dotnet/roslyn/issues/51062#issuecomment-775749582
that I have not had a successful approach to interacting on these topics.

I think you have a much better chance. 🤞